polytracker/src/taintdag/fdmapping.cpp

```diff
@@ -1,5 +1,28 @@
 #include "taintdag/fdmapping.hpp"
 
+#include <iterator>
+
+namespace {
+
+// Helper structure that allows range-based for iteration
+// on the entries in FDMapping.
+template <typename It> struct RetVal {
```

Comment:
We could maybe use [llvm::make_range](https://llvm.org/doxygen/iterator__range_8h.html) here?

Reply:
True. It is almost identical implementation, they handle more cases (move etc.). I don't think we have any dependency towards llvm in the runtime part of polytracker. But this is header only and is probably a good idea. Will have a look.

Reply:
> I don't think we have any dependency towards llvm in the runtime part of polytracker.

Ah, okay, didn't realize this! I think we should keep it that way then.

Reply:
We can't use it right away, would still need some code to map it into a reverse iterator. Keep as is?

Reply:
Yes.